### PR TITLE
Adds values for existing income dets

### PR DESCRIPTION
**Pull request by jlhogan**
wondering how others feel about the merits of having validation errors around "incomeNotDisclosed" ? I don't know that it actually is useful, but maybe it is? happy to put it back, since this change to the DML prevents the bad state

### CODE REVIEW

File: src/migration/1608582587797-AddIncomeNotDisclosedToFamily.ts [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+import { dropDefaultConstraint } from './utils/dropDefaultConstraint';
 
 export class AddIncomeNotDisclosedToFamily1608582587797
   implements MigrationInterface {
   name = 'AddIncomeNotDisclosedToFamily1608582587797';
 
   public async up(queryRunner: QueryRunner): Promise<void> {
     await queryRunner.query(
-      `ALTER TABLE "income_determination" ADD "incomeNotDisclosed" bit`
+      `ALTER TABLE "income_determination" ADD "incomeNotDisclosed" bit DEFAULT(0) WITH VALUES`
```
Comment (JordanGuinn) [inline src/migration/1608582587797-AddIncomeNotDisclosedToFamily.ts:10]: Should we default to `false` only here (for historical records), or in the entity as well (for future records)?

**Review by JordanGuinn — APPROVED** [review]
I think I agree - the value of `incomeNotDisclosed` seems to come when paired with the presence/absence of an actual income, not on its own.

**Review by bamader — APPROVED** [review]
Jordan said it perfectly - the field on its own doesn't help much, it's the pairing with other income fields that's informative.

**Comment by mmazanec22** [comment]
Instead of a validation we could have a test that validates that the other fields are cleared when it is checked

File: src/migration/1608582587797-AddIncomeNotDisclosedToFamily.ts (hunk 1 above)
Comment (jlhogan) [reply src/migration/1608582587797-AddIncomeNotDisclosedToFamily.ts:10]: i dont think there should be a default for future records -- if they're created wthout the explicit value, then the user didn't add the specific value and we need to prompt them to add it or explicitly say "not disclosed" (at least that's the pattern we've gone with for other fields that are potentially not collected)

**Pull request merged**